src/Modal/ModalContents.js

```diff
@@ -97,7 +99,8 @@ class ModalContents extends Util.mixin(BindMixin) {
       }
 
       let maxHeightDifference = maxHeight - prevMaxHeight;
-      if (maxHeightDifference > 0) {
+      if (maxHeightDifference > 0 &&
```

Comment:
All other changes are simple for resizing performance improvements. This logic right here is what stops the modal from getting huge